Description: Use Debian filesystem locations for FreeWnn data.
 Install read-only data below datadir/wnn, mutable dictionaries below
 localstatedir/lib/wnn, and install files as root while retaining each
 language-specific mutable dictionary directory.
 Historical packaging reference: 82eaf4f6daf49601c2f23cda13c0f868ac458fe3.
 Source commits: no upstream source commit.
Origin: vendor
Bug-Debian: https://bugs.debian.org/699622
Forwarded: not-needed
Last-Update: 2026-07-14

--- a/makerule.mk.in
+++ b/makerule.mk.in
@@ -126,7 +126,7 @@ LDOPTIONS = $(LDFLAGS) @CDEBUGFLAGS@ $(C
 # OMRON INPUT METHOD MAKE ENV
 #
 
-       WNNOWNER = wnn
+       WNNOWNER = root
        UUMOWNER = @UUMOWNER@
        UUMGROUP = @UUMGROUP@
 
@@ -136,7 +136,8 @@ LDOPTIONS = $(LDFLAGS) @CDEBUGFLAGS@ $(C
     exec_prefix = @exec_prefix@
          bindir = @bindir@
         sbindir = @sbindir@
-      WNNWNNDIR = $(exec_prefix)/lib/wnn
+      WNNWNNDIR = $(datadir)/wnn
+      WNNVARDIR = $(localstatedir)/lib/wnn
       WNNBINDIR = $(bindir)
      WNNSBINDIR = @WNNSBINDIR@
       WNNINCDIR = @includedir@
@@ -158,11 +159,12 @@ LDOPTIONS = $(LDFLAGS) @CDEBUGFLAGS@ $(C
 
          WNNTOP = $(IMTOP)/Wnn
        JWNNLANG = ja_JP
+     JWNNVARDIR = $(WNNVARDIR)/$(JWNNLANG)
      JWNNWNNDIR = $(WNNWNNDIR)/$(JWNNLANG)
      JWNNBINDIR = $(WNNBINDIR)$(JWNNSUBBINDIR)
     JWNNSBINDIR = $(WNNSBINDIR)$(JWNNSUBBINDIR)
      JWNNINCDIR = $(WNNINCDIR)/wnn
-     JWNNDICDIR = $(JWNNWNNDIR)/dic
+     JWNNDICDIR = $(JWNNVARDIR)/dic
   JWNNPUBDICDIR = $(JWNNDICDIR)/pubdic
 JWNNWNNCONSDICDIR = $(JWNNDICDIR)/wnncons
   JWNNFZKSRCDIR = $(JWNNDICDIR)/src
@@ -189,11 +191,12 @@ WNNWNNCONSDICSRC = $(WNNWNNCONSSRC)/dic
 
         CWNNTOP = $(IMTOP)/cWnn
        CWNNLANG = zh_CN
+     CWNNVARDIR = $(WNNVARDIR)/$(CWNNLANG)
      CWNNWNNDIR = $(WNNWNNDIR)/$(CWNNLANG)
      CWNNINCDIR = $(WNNINCDIR)/cwnn
      CWNNBINDIR = $(WNNBINDIR)$(CWNNSUBBINDIR)
     CWNNSBINDIR = $(WNNSBINDIR)$(CWNNSUBBINDIR)
-     CWNNDICDIR = $(CWNNWNNDIR)/dic
+     CWNNDICDIR = $(CWNNVARDIR)/dic
   CWNNSYSDICDIR = $(CWNNDICDIR)/sys
       CWNNCDSRC = $(CWNNTOP)/cd
     CWNNCDICSRC = $(CWNNTOP)/cdic
@@ -213,11 +216,12 @@ CWNNJLIBVERSION = 0:0
 
         KWNNTOP = $(IMTOP)/kWnn
        KWNNLANG = ko_KR
+     KWNNVARDIR = $(WNNVARDIR)/$(KWNNLANG)
      KWNNWNNDIR = $(WNNWNNDIR)/$(KWNNLANG)
      KWNNINCDIR = $(WNNINCDIR)/kwnn
      KWNNBINDIR = $(WNNBINDIR)$(KWNNSUBBINDIR)
     KWNNSBINDIR = $(WNNSBINDIR)$(KWNNSUBBINDIR)
-     KWNNDICDIR = $(KWNNWNNDIR)/dic
+     KWNNDICDIR = $(KWNNVARDIR)/dic
   KWNNSYSDICDIR = $(KWNNDICDIR)/sys
       KWNNKDSRC = $(KWNNTOP)/kd
     KWNNKDICSRC = $(KWNNTOP)/kdic
@@ -236,8 +240,9 @@ KWNNJLIBVERSION = 0:0
     KWNNCONVLIB = $(KWNNCONVSRC)/libconvkey.a
 
        TWNNLANG = zh_TW
+     TWNNVARDIR = $(WNNVARDIR)/$(TWNNLANG)
      TWNNWNNDIR = $(WNNWNNDIR)/$(TWNNLANG)
-     TWNNDICDIR = $(TWNNWNNDIR)/dic
+     TWNNDICDIR = $(TWNNVARDIR)/dic
   TWNNSYSDICDIR = $(TWNNDICDIR)/sys
       TWNNTDSRC = $(CWNNTOP)/td
     TWNNTDICSRC = $(CWNNTOP)/tdic
